Add qint16 support via int16

axitkhurana opened the pull request:
Summary:
Pytorch doesn't have support for qint16 yet. Add an option to handle qint16 via int16 & qint32 data types.

* For qint16 tensors in NNAPI, the user sends a qint32 tensor. We convert the qint32 to int16 for the converter and set the zero point and scale for nnapi
   * inputs to the model have to have fixed scale and zero point and are only supported for testing
* Added a flag `use_int16_for_qint16` which will be used maintain backwards compatibility in the converter when true qint16 is supported in PyTorch

Test Plan: pytest test/test_nnapi.py

Reviewed By: dreiss

Differential Revision: D33285124

jerryzh168 commented:
hmmm, we should add official qint16 support, like in https://github.com/pytorch/pytorch/pull/44678

Commit:
nnapi: Add int32 type torchscript expressions (#70197)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/70197

Test Plan:
* `pytest test/test_nnapi.py`
* Testing via ops following this commit

Differential Revision: D33237917

fbshipit-source-id: a5ea8e4050d7f72e2c95cb928c574a773399ac87

Commit:
NNAPI: Add runtime flexible shapes & return shapes (#70334)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/70334

* Use 0 for load time flexible shapes
* -1 for runtime flexible shapes
* NNAPI needs return shapes for flexible outputs

Test Plan: Tested via upcoming ops

Differential Revision: D33237922

fbshipit-source-id: fd2e32ecd81f91a3977a3e096de0b16a173eef45

Commit:
Add qint16 support via int16 (#70376)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/70376

Pytorch doesn't have support for qint16 yet. Add an option to handle qint16 via int16 & qint32 data types.

* For qint16 tensors in NNAPI, the user sends a qint32 tensor. We convert the qint32 to int16 for the converter and set the zero point and scale for nnapi
   * inputs to the model have to have fixed scale and zero point and are only supported for testing
* Added a flag `use_int16_for_qint16` which will be used maintain backwards compatibility in the converter when true qint16 is supported in PyTorch

Test Plan: pytest test/test_nnapi.py

Reviewed By: dreiss

Differential Revision: D33285124

fbshipit-source-id: 13a79fe8312ed9d55db5c5441474ad5dcfe0489a
